test/test_torch.py

```diff
@@ -16656,6 +16656,39 @@ def test_dot(self, device, dtype):
         torch.dot(v1, v2, out=out)
         self.assertEqual(res1, out)
 
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_vdot(self, device, dtype):
```

Comment:
Can this unify test_dot and test_vdot by creating two tests that call a common helper?

```
_test_dot_vdot_helper(self, device, dtype, op, ref):
   ...

test_dot(...):
  ...

test_vdot(...):
  ...
```

Where both tests call `_test_dot_vdot_helper`. This would also test our current dot implementation vs. NumPy's dot, which would be nice, and get it testing on CUDA, which would also be good.

Reply:
Makes sense. Was thinking of doing the same but since `dot` has an `out` variant the testing would have slightly diverged so did not touch `dot`'s tests.

Reply:
Done.